.github/workflows/build-deps.yml

```diff
@@ -84,63 +88,156 @@ jobs:
-          brew remove --force --ignore-dependencies $(brew list --formula)
-          brew install coreutils
-
-      - name: Set up Python
-        env:
-          # Despite the name, this is built for the macOS 11 SDK on arm64 and 10.9+ on intel
-          PYTHON3_DOWNLOAD_URL: "https://www.python.org/ftp/python/3.11.5/python-3.11.5-macos11.pkg"
-          PYTHON2_DOWNLOAD_URL: "https://www.python.org/ftp/python/2.7.18/python-2.7.18-macosx10.9.pkg"
-        run: |
-          curl "$PYTHON3_DOWNLOAD_URL" -o python3.pkg
-          sudo installer -pkg python3.pkg -target /
-
-          curl "$PYTHON2_DOWNLOAD_URL" -o python2.pkg
-          sudo installer -pkg python2.pkg -target /
-
-      - name: Checkout code
-        uses: actions/checkout@v4
-
-      - name: Run the build
-        env:
-          DD_PYTHON3: "/Library/Frameworks/Python.framework/Versions/3.11/bin/python3"
-          DD_PYTHON2: "/Library/Frameworks/Python.framework/Versions/2.7/bin/python"
-          # This sets the minimum mac os version compatible for all built artifacts
-          MACOSX_DEPLOYMENT_TARGET: "10.12"
-        run: |
-          ${DD_PYTHON3} -m pip install -r .builders/deps/host_dependencies.txt
-
-          mkdir builder_root
-          ${DD_PYTHON3} .builders/build.py --builder-root builder_root --python 3 out_py3
-          ${DD_PYTHON3} .builders/build.py --builder-root builder_root --skip-setup --python 2 out_py2
+    - name: Checkout code
+      uses: actions/checkout@v4
+      with:
+        ref: "${{ github.head_ref }}"
+
+    - name: Set up Python ${{ env.PYTHON_VERSION }}
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ env.PYTHON_VERSION }}
+
+    - name: Install management dependencies
+      run: pip install -r .builders/deps/host_dependencies.txt
+
+    - name: Download artifacts
+      uses: actions/download-artifact@v4
+      with:
+        path: targets
+        pattern: target-*
+        merge-multiple: true
+
+    - name: Get credentials
+      id: auth
+      uses: google-github-actions/auth@v2
+      with:
+        project_id: datadog-agent-int-build
+        workload_identity_provider: projects/574011472402/locations/global/workloadIdentityPools/github/providers/integrations-core
+
+    - name: Upload dependencies and update lock files
+      run: |-
+        mkdir -p .deps/resolved
+        rm -rf .deps/resolved/*
+
+        cd targets
+        for target in *; do
+          cd $target
+
+          for pyver in *; do
+            python ${{ github.workspace }}/.builders/upload.py $pyver/wheels
+          done
+          mv $pyver/frozen.txt ${{ github.workspace }}/.deps/resolved/${target}_${pyver}.txt
+
+          cd ..
+        done
+        cd ..
+        rm -rf targets
+
+        hash=$(openssl dgst -sha256 "${{ env.DIRECT_DEPENDENCY_FILE }}" | awk '{ print $2 }')
+        echo "{\"sha256\": \"$hash\"}" > ".deps/metadata.json"
```

Comment:
Any reason not to do this in Python as well?

Reply:
No reason I suppose, I'll switch!